### PR TITLE
feat(create/delete_lease): add auth middleware

### PR DESCRIPTION
**Pull request by jchauncey**
for authorizing requests on handlers requiring it

### CODE REVIEW

File: handlers/delete_lease_test.go [hunk 1]
```diff
@@ -63,13 +67,16 @@ func TestDeleteLeaseNoSuchLease(t *testing.T) {
...
+	req.Header.Set("Authorization", "some awesome token")
 	assert.NoErr(t, err)
 	res := httptest.NewRecorder()
 	hdl.ServeHTTP(res, req)
 	assert.Equal(t, res.Code, http.StatusConflict, "response code")
 }
 
 func TestDeleteLeaseExists(t *testing.T) {
+	os.Setenv("AUTH_TOKEN", "some awesome token")
+	defer os.Clearenv()
```
Comment (arschles) [inline handlers/delete_lease_test.go]: @jchauncey is this env var still required?

File: main.go [hunk 2]
```diff
@@ -15,11 +15,11 @@ const (
...
 )
 
-func configureRoutes(serveMux *http.ServeMux, createLeaseHandler http.Handler, deleteLeaseHandler http.Handler) {
+func configureRoutesWithAuth(serveMux *http.ServeMux, createLeaseHandler http.Handler, deleteLeaseHandler http.Handler, authToken string) {
 	createLeaseHandler = htp.MethodMux(map[htp.Method]http.Handler{htp.Post: createLeaseHandler})
 	deleteLeaseHandler = htp.MethodMux(map[htp.Method]http.Handler{htp.Delete: deleteLeaseHandler})
-	serveMux.Handle("/lease", createLeaseHandler)
-	serveMux.Handle("/lease/", deleteLeaseHandler)
+	serveMux.Handle("/lease", handlers.WithAuth(authToken, "Authorization", createLeaseHandler))
+	serveMux.Handle("/lease/", handlers.WithAuth(authToken, "Authorization", deleteLeaseHandler))
```
Comment (arschles) [inline main.go]: can you put `"Authorization"` into a `const`?

File: handlers/delete_lease_test.go (hunk 1 above)
Comment (jchauncey) [inline handlers/delete_lease_test.go]: whoops no

**Comment by arschles** [comment]
Code LGTM, assuming https://github.com/deis/k8s-claimer/pull/52#discussion_r60990015 and https://github.com/deis/k8s-claimer/pull/52#discussion_r60990113 (both small nits) are addressed

**Comment by arschles** [comment]
Replaces #51 
Closes #17

**Pull request merged**